.github/workflows/lint.yaml

```diff
@@ -6,34 +6,16 @@ on:
       - main
   pull_request:
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name == 'workflow_dispatch' }}
+  cancel-in-progress: true
+
 jobs:
   optional-lint:
     name: Optional Lint
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Setup Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: "3.10"
-      # Install onnx so that it is found by the linters
-      - name: Install ONNX
-        run: |
-          source workflow_scripts/protobuf/build_protobuf_unix.sh $(nproc)
-
-          python -m pip install --quiet --upgrade pip setuptools wheel
-          python -m pip install --quiet -r requirements-release.txt
-
-          sudo apt-get install -qq -o=Dpkg::Use-Pty=0 -y --no-install-recommends dos2unix
-          git submodule update --init --recursive
-          export ONNX_BUILD_TESTS=0
-          export ONNX_ML=1
-          export CMAKE_ARGS="-DONNXIFI_DUMMY_BACKEND=ON -DONNX_WERROR=ON"
-          # ONNX lite
-          export CMAKE_ARGS="${CMAKE_ARGS} -DONNX_USE_LITE_PROTO=ON"
-          export ONNX_NAMESPACE=ONNX_NAMESPACE_FOO_BAR_FOR_CI
-
-          python setup.py install
```

Comment:
To confirm, we don't need to build ONNX from source in the Lint CI anymore?

Reply:
Yes for the optional lints. Lintrunner still needs built onnx

Reply:
https://github.com/charliermarsh/ruff#target-version